GoMeet/src/main/java/com/google/sps/servlets/EmailServlet.java

```diff
@@ -0,0 +1,79 @@
+package com.google.sps.servlets;
+
+import com.google.sps.data.ErrorMessages;
+import com.google.sps.data.ServletUtil;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Properties;
+import javax.mail.internet.AddressException;
+import javax.mail.internet.InternetAddress;
+import javax.mail.internet.MimeMessage;
+import javax.mail.Message;
+import javax.mail.MessagingException;
+import javax.mail.Session;
+import javax.mail.Transport;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/email")
+public class EmailServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    
+    String meetingEventId = request.getParameter("meetingEventId"); 
+    String guestListUri = request.getParameter("guestList");
+
+    if (meetingEventId == null || guestListUri == null) {
+      ServletUtil.sendErrorResponse(response, HttpServletResponse.SC_BAD_REQUEST, 
+          ErrorMessages.BAD_EMAIL_REQUEST_ERROR);
+      return; 
+    }
+
+    String guestList = ServletUtil.decodeUri(guestListUri); 
+    String[] guestListSplit = guestList.split(",");
+    for (int i = 0; i < guestListSplit.length; i++) {
+      boolean sent = sendEmail(guestListSplit[i], meetingEventId);
+      if (!sent) {
+        ServletUtil.sendErrorResponse(response, HttpServletResponse.SC_BAD_REQUEST, 
+            ErrorMessages.BAD_EMAIL_REQUEST_ERROR);
+        return; 
```

Comment:
Would it be better to keep trying to send it to the remaining emails in the list instead of returning immediately?

Reply:
This is a good point. Either approach is reasonable, but we should at least add a comment here, or in the design, as to why we chose how to handle this error.

Reply:
I have changed the design to continue calling the send email function - the json response is now a hashmap with each email address as a key and the corresponding sent status (true/false) as the value. I will probably use this response data to display a message on the html page e.g. "Emails sent successfully to X, Y and Z. Unable to send emails to A and B."

Reply:
Sounds good to me :)